File: add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        n1 = 0
        n2 = 0
        factor = 1
        while l1 is not None:
            n1 = n1 + l1.val * factor
            l1 = l1.next
            factor = factor * 10
            
        factor = 1
        
        while l2 is not None:
            n2 = n2 + l2.val * factor
            l2 = l2.next
            factor = factor * 10
        
        n = n1 + n2
        if(n is 0):
            return ListNode(0)
        res = ListNode(0)
        head = res
        while n != 0:
            res.next = ListNode(n % 10)
            n = n/10
            res = res.next
        return head.next

File: add-two-numbers/add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int carry = 0;
        ListNode head = null;
        ListNode prev = null;
        while(l1!=null && l2!=null){
            ListNode curr = new ListNode((l1.val+l2.val+carry)%10);
            carry = (l1.val+l2.val+carry)/10;
            if(prev!=null)prev.next = curr;
            prev = curr;
            l1 = l1.next;
            l2=l2.next;
            if(head==null)head=curr;
        }
        while(l1!=null){
            ListNode curr = new ListNode((l1.val+carry)%10);
            carry = (l1.val+carry)/10;
            if(prev!=null)prev.next = curr;
            prev = curr;
            l1 = l1.next;
            if(head==null)head=curr;
        }
        while(l2!=null){
            ListNode curr = new ListNode((l2.val+carry)%10);
            carry = (l2.val+carry)/10;
            if(prev!=null)prev.next = curr;
            prev = curr;
            l2 = l2.next;
            if(head==null)head=curr;
        }
        if(carry!=0){
            ListNode curr = new ListNode(carry);
            prev.next = curr;
        }
        return head;
    }
}